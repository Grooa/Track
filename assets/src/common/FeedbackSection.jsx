import PropTypes from 'prop-types';

const FeedbackSection = (props) => {
  const {
    title = null,
    text = null,
    link = null,
    type = null,
  } = props;

  let cssClasses = 'feedback';
  if (type) {
    cssClasses += ` feedback-${type}`;
  }

  const textRendered = text ? <p>{text}</p> : '';
  const linkRendered = link ? <a href={link.url} className="button pill">{link.label}</a> : '';

  if (!title) {
    return <div className={cssClasses}>
      {textRendered}

      {linkRendered}
    </div>;
  }

  return <section className={cssClasses}>
    <h2>{title}</h2>

    {textRendered}

    {linkRendered}
  </section>;
};

FeedbackSection.propTypes = {
  title: PropTypes.string,
  text: PropTypes.string,
  link: PropTypes.object,
  type: PropTypes.string,
};

export default FeedbackSection;
