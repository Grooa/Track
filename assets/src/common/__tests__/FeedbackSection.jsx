import { render, shallow } from 'enzyme';

import FeedbackSection from '../FeedbackSection';

describe('<FeedbackSection />', () => {
  let title;
  let text;
  let type;
  let link;

  beforeEach(() => {
    title = 'Hello world';
    text = 'Lorem lipsum dolor sit amed';
    type = 'error';
    link = { url: 'https://test.com/test', label: 'Test button' };
  });

  it('should render empty block properly', () => {
    const wrapper = render(<FeedbackSection/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render with only prop `title` properly', () => {
    const wrapper = render(<FeedbackSection title={title}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render with only prop text properly', () => {
    const wrapper = render(<FeedbackSection text={text}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render with only prop link properly', () => {
    const wrapper = render(<FeedbackSection link={link}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render full block properly', () => {
    const wrapper = render(<FeedbackSection
      title={title}
      text={text}
      type={type}
      link={link}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render div when prop title is empty', () => {
    const wrapper = shallow(<FeedbackSection/>);

    expect(wrapper.find('section').length).toBe(0);
    expect(wrapper.find('div').length).toBe(1);
  });

  it('should render section when prop title is not empty', () => {
    const wrapper = shallow(<FeedbackSection title={title}/>);

    expect(wrapper.find('div').length).toBe(0);
    expect(wrapper.find('section').length).toBe(1);
  });

  it('should set prop `type` as className `feedback-{type}` on <div>', () => {
    const wrapper = shallow(<FeedbackSection type={type}/>);

    expect(wrapper.find('div').hasClass(`feedback-${type}`)).toBeTruthy();
  });

  it('should set prop `type` as className `feedback-{type}` on <section>', () => {
    const wrapper = shallow(<FeedbackSection title={title} type={type}/>);

    expect(wrapper.find('section').hasClass(`feedback-${type}`)).toBeTruthy();
  });
});
