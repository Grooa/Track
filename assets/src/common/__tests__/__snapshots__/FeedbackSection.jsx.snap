// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeedbackSection /> should render empty block properly 1`] = `
<div
  class="feedback"
/>
`;

exports[`<FeedbackSection /> should render full block properly 1`] = `
<section
  class="feedback feedback-error"
>
  <h2>
    Hello world
  </h2>
  <p>
    Lorem lipsum dolor sit amed
  </p>
  <a
    class="button pill"
    href="https://test.com/test"
  >
    Test button
  </a>
</section>
`;

exports[`<FeedbackSection /> should render with only prop \`title\` properly 1`] = `
<section
  class="feedback"
>
  <h2>
    Hello world
  </h2>
</section>
`;

exports[`<FeedbackSection /> should render with only prop link properly 1`] = `
<div
  class="feedback"
>
  <a
    class="button pill"
    href="https://test.com/test"
  >
    Test button
  </a>
</div>
`;

exports[`<FeedbackSection /> should render with only prop text properly 1`] = `
<div
  class="feedback"
>
  <p>
    Lorem lipsum dolor sit amed
  </p>
</div>
`;
